src/generic/grideditors.cpp

```diff
@@ -100,7 +93,7 @@ void wxGridCellEditorEvtHandler::OnKeyDown(wxKeyEvent& event)
     {
         case WXK_ESCAPE:
             m_editor->Reset();
-            m_grid->DisableCellEditControl();
+            m_grid->CallAfter(&wxGrid::DisableCellEditControl);
```

Comment:
Two reasons: consistency with the other calls and the fact that the comment in the other event handler says that directly calling this function is not good because when the editor is disabled the event handler could be destroyed - leading to invalid memory access (that was why the kill focus handler used the event to request the closing of the editor). This call is also inside an event handler, so it would run into the same issue of the event handler object being destroyed before this event is finished processing. The case where a crash like this could happen is probably rare, since it would require pointers to be overwritten in a very short timespan, but since we already know how to fix it, I think it is better to fix it before someone reports a crash with it (or has to try and track down a crash in it).

Reply:
Sometimes postponing the event can result in unexpected problems too, e.g. if a handler for some event running immediately after this `wxEVT_KEY_DOWN` relies on the editor being already dismissed, but I guess it's indeed better for consistency, so let's do it like this and see if anybody complains...